### ISSUE TITLE
Fixed height in "change height" / "random height" tool window doesn't get updated when changed in the other one

### ISSUE THREAD

**Issue by bunnybot** [widelands/widelands#6969]
[<img align='left' src='https://codeberg.org/avatars/593dc43e2444c25db9481960e5b693ab' alt='tothxa' height='48' />](https://codeberg.org/tothxa)*[Mirrored from Codeberg](https://codeberg.org/wl/widelands/issues/5329)*
*Created on Wed Jan 07 23:43:04 CET 2026 by [**Tóth András (tothxa)**](https://codeberg.org/tothxa)*

---

### Describe the Bug

The fixed height of the height tool and the random height tool are actually shared, but if you have both windows open and change the fixed height in one then it's not updated in the other one. This means that if you then switch to the other one and ctrl-click on the map, the height will be set to the one that was last changed, even though the active window still shows the old fixed height.

### To Reproduce

1. Start the editor
2. Open the height tool
3. Open the random height tool
4. Increase the fixed height in the random height window by several steps to make the difference well visible
5. "Draw" a few fields with the new height with CTRL held down
6. Switch to the height tool: note that the fixed height still shows the old value
7. "Draw" a few fields with the new height with CTRL held down
8. See error: the new height is used
9. Close and reopen the height window
10. See that it now shows the new height (proof that it uses the same variable)

### Expected Behavior

The fixed height should be updated in the other window too when changed.

Alternatively use separate fixed height variables for the two tools, but this is probably more intrusive and not important.

### Operating System

antiX/Debian trixie

### Widelands Version

v1.3

### Enabled Add-Ons

n/a
